src/main/java/servlets/ReceiptAnalysis.java

```diff
@@ -56,6 +63,11 @@
   // minimum confidence score that a detected logo must have to be considered significant for
   // receipt analysis.
   private static final float LOGO_DETECTION_CONFIDENCE_THRESHOLD = 0.4f;
+  // Matches strings containing at least one digit.
+  private static final Pattern digitRegex = Pattern.compile(".*\\d.*");
+  // Matches strings in U.S. date format.
+  private static final Pattern dateRegex =
+      Pattern.compile("\\d?\\d([/-])\\d?\\d\\1\\d{2}(\\d{2})?");
```

Comment:
Eventually, a more specific regex should be used here because the month and day of a date are restricted to ranges, and this would match an invalid date like 13/32/2020.